### PR TITLE
Feature/opt out message

### PR DESCRIPTION
**Pull request by teehamaral**
https://app.asana.com/0/1201638023907740/1201674230141709

### CODE REVIEW

**Review by musamusa — APPROVED** [review]

File: backends/rapidpro/backend.go [hunk 1]
```diff
@@ -156,6 +156,11 @@ func (b *backend) NewOutgoingMsg(channel courier.Channel, urn urns.URN, text str
 	return newMsg(MsgOutgoing, channel, urn, text)
 }
 
+// NewMsgOutgoing creates a new outgoing message from the given params
+func (b *backend) NewMsgOutgoing(channel courier.Channel, urn urns.URN, text string) courier.Msg {
```
Comment (ihor-palii) [inline backends/rapidpro/backend.go:160]: We already have the `NewOutgoingMsg` method with the same logic. So I guess this method can be removed.

File: backend.go [hunk 2]
```diff
@@ -44,6 +44,9 @@ type Backend interface {
 	// NewIncomingMsg creates a new message from the given params
 	NewIncomingMsg(channel Channel, urn urns.URN, text string) Msg
 
+	// NewMsgOutgoing creates a new outgoing message from the given params
+	NewMsgOutgoing(channel Channel, urn urns.URN, text string) Msg
```
Comment (ihor-palii) [inline backend.go:48]: Need to be replaced with `NewOutgoingMsg` that was already implemented.

File: handlers/twiml/twiml.go [hunk 3]
```diff
@@ -136,6 +136,17 @@ func (h *handler) receiveMessage(ctx context.Context, channel courier.Channel, w
 	// build our msg
 	msg := h.Backend().NewIncomingMsg(channel, urn, form.Body).WithExternalID(form.MessageSID).WithSegmentsCount(form.NumSegments)
 
+	// check whether it's a short code and the text contains opt-out words
+	if len(channel.Address()) <= 6 && utils.CheckOptOutKeywordPresence(msg.Text()) {
+		textMessage := channel.OrgConfigForKey(utils.OptOutMessageBackKey, utils.OptOutDefaultMessageBack)
+		msgBack := h.Backend().NewMsgOutgoing(channel, urn, textMessage.(string))
```
Comment (ihor-palii) [inline handlers/twiml/twiml.go:142]: Replace `NewMsgOutgoing` with `NewOutgoingMsg`.

File: test.go [hunk 4]
```diff
@@ -128,6 +128,11 @@ func (mb *MockBackend) NewOutgoingMsg(channel Channel, id MsgID, urn urns.URN, t
 	return &mockMsg{channel: channel, id: id, urn: urn, text: text, highPriority: highPriority, quickReplies: quickReplies, topic: topic, responseToID: msgResponseToID, responseToExternalID: responseToExternalID}
 }
 
+// NewMsgOutgoing creates a new outgoing message from the given params
+func (mb *MockBackend) NewMsgOutgoing(channel Channel, urn urns.URN, text string) Msg {
```
Comment (ihor-palii) [inline test.go:132]: Can de removed as ve have `NewOutgoingMsg`

**Review by ihor-palii — CHANGES_REQUESTED** [review]

**Review by ihor-palii — APPROVED** [review]

**Pull request merged**